docs/iot/tutorials/gpio-input.md

```diff
@@ -0,0 +1,138 @@
+    - `RegisterCallbackForPinValueChangedEvent()` registers an anonymous callback function for both the `PinEventTypes.Rising` and `PinEventTypes.Falling` events on the pin. These events correspond to pin states of `PinValue.High` and `PinValue.Low`, respectively.
+    - The callback function delegates to a method called `onPinEvent`. `onPinEvent` uses another ternary expression that also writes the corresponding `alert` or `ready` strings.
+    - The main thread sleeps indefinitely while waiting for pin events.
+
+1. [!INCLUDE [tutorial-build](../includes/tutorial-build.md)]
+1. [!INCLUDE [tutorial-deploy](../includes/tutorial-deploy.md)]
+1. Run the app on the Raspberry Pi by switching to the deployment directory and running the executable.
+
+    ```bash
+    ./InputTutorial
+    ```
+
+    The app displays text similar to the following:
+
+    ```console
+    Initial status (05/10/2022 15:59:25): Ready ✅
+    ```
+
+1. Disconnect pin 21 from ground. The console displays text similar to the following:
+
+    ```console
+    (05/10/2022 15:59:59) ALERT! 🚨
+    ```
+
+1. Reconnect pin 21 and ground. The console displays text similar to the following:
+
+    ```console
+    (05/10/2022 16:00:25) Ready ✅
+    ```
+
+1. Terminate the program by pressing <kbd>Ctrl</kbd>+<kbd>C</kbd>.
+
+Congratulations! You've used GPIO to detect input! There are many uses for this type of input. This example can be used with any scenario where a switch connects or breaks a circuit. Here's an example using it with a magnetic reed switch, which is often used to detect open doors or windows.
+
+:::image type="content" source="https://via.placeholder.com/600x400.png" alt-text="Animated GIF showing magnetic reed switch":::
+
+## Laser tripwire
+
+A similar application involves laser tripwires. Building a laser tripwire requires the following additional components:
+
+* KY-008 laser transmitter module
+* Laser receiver sensor module *(see note below)*
+* 10K Ω resistor
+* 20K Ω resistor
+
+> [!NOTE]
+> *Laser receiver sensor module* is the generic name applied to a common module found at many internet retailers. The device may vary in name or manufacturer, but should resemble this image.
+>
+> ![Image of a Laser Receiver Sensor Module](https://via.placeholder.com/200x200.png)
+
+### Hardware
+
+Connect the components as detailed in following diagram.
+
+:::image type="content" source="https://via.placeholder.com/500x300.png" alt-text="Fritzing diagram":::
+
+Pay special attention to the 10K Ω and 20K Ω resistors. These implement a [voltage divider](https://en.wikipedia.org/wiki/Voltage_divider). This is because the laser receiver module outputs 5V to indicate the beam is broken. Raspberry Pi only supports up to 3.3V for GPIO input. Since sending the full 5V to the pin could damage the Raspberry Pi, the current from the receiver module is passed through a voltage divider to halve the voltage to 2.5V.
```

Comment:
```suggestion
Pay close attention to the 10K Ω and 20K Ω resistors. These implement a [voltage divider](https://en.wikipedia.org/wiki/Voltage_divider). This is because the laser receiver module outputs 5V to indicate the beam is broken. Raspberry Pi only supports up to 3.3V for GPIO input. Since sending the full 5V to the pin could damage the Raspberry Pi, the current from the receiver module is passed through a voltage divider to halve the voltage to 2.5V.
```

Also, I heard a few peers mention in the past that we should try to avoid Wikipedia links as the validity is unverified. Instead, consider creating a .NET IoT glossary article, and writing your own interpretation of these sorts of things.